staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go

```diff
@@ -285,6 +285,9 @@ type ValidationRule struct {
 	// e.g. for attribute `foo.34$` appears in a list `testList`, the fieldPath could be set to `.testList['foo.34$']`
 	// +optional
 	FieldPath string `json:"fieldPath,omitempty" protobuf:"bytes,5,opt,name=fieldPath"`
+
+	// +optional
+	OptionalOldSelf *bool `json:"optionalOldSelf,omitempty"`
```

Comment:
Is the field itself feature gated? The name is confusing. Since it is added into a stable API, it should follow the feature gate policy maybe?

Reply:
The field only does anything if the feature gate is enabled. Is that what you mean? Or you mean it should throw a validation error if gate is not enabled?

Reply:
I mean adding the field itself might also need to be feature gated

Reply:
So the behavior should be: the newly added field will be ignored and hidden from user if the feature gate is disabled. And ignore the existing old data if you see while updating with feature gate disabled. And add the detailed doc on this new field(like what it did, which feature gate it's controlled, etc)

Reply:
Added validation and godoc with feature gate info PTAL